diff --git a/libcontainerd/remote_linux.go b/libcontainerd/remote_linux.go
index 12ce0e1..4918109 100644
--- a/libcontainerd/remote_linux.go
+++ b/libcontainerd/remote_linux.go
@@ -33,6 +33,7 @@ const (
 	containerdBinary          = "docker-containerd"
 	containerdPidFilename     = "docker-containerd.pid"
 	containerdSockFilename    = "docker-containerd.sock"
+	containerdStateDir        = "containerd"
 	eventTimestampFilename    = "event.ts"
 )
 
@@ -349,7 +350,7 @@ func (r *remote) runContainerdDaemon() error {
 	}
 
 	// Start a new instance
-	args := []string{"-l", r.rpcAddr, "--runtime", "docker-runc", "--start-timeout", "2m"}
+	args := []string{"-l", r.rpcAddr, "--runtime", "docker-runc", "--start-timeout", "2m", "--state-dir", filepath.Join(r.stateDir, containerdStateDir)}
 	if r.debugLog {
 		args = append(args, "--debug", "--metrics-interval=0")
 	}
