diff --git a/profiles/apparmor/apparmor.go b/profiles/apparmor/apparmor.go
index 51dfa5c..4def938 100644
--- a/profiles/apparmor/apparmor.go
+++ b/profiles/apparmor/apparmor.go
@@ -16,11 +16,23 @@ import (
 
 var (
 	// profileDirectory is the file store for apparmor profiles and macros.
-	profileDirectory = "/etc/apparmor.d"
+	profileDirectory = defaultProfileDirectory()
 )
 
+func defaultProfileDirectory() string {
+	// Ubuntu Snappy has read-only /etc/apparmor.d
+	if snapData := os.Getenv("SNAP_DATA"); snapData != "" {
+		// Ubuntu Snappy 16.04+
+		return snapData + "/etc/apparmor.d"
+	} else if os.Getenv("SNAP_APP_DATA_PATH") != "" {
+		// Ubuntu Snappy 15.04+
+		return "/var/lib/apparmor/profiles"
+	}
+	return "/etc/apparmor.d"
+}
+
 // profileData holds information about the given profile for generation.
 type profileData struct {
 	// Name is profile name.
