diff --git a/daemon/apparmor_default.go b/daemon/apparmor_default.go
index 2a418b2..278f320 100644
--- a/daemon/apparmor_default.go
+++ b/daemon/apparmor_default.go
@@ -4,6 +4,7 @@ package daemon
 
 import (
 	"fmt"
+	"os"
 
 	aaprofile "github.com/docker/docker/profiles/apparmor"
 	"github.com/opencontainers/runc/libcontainer/apparmor"
@@ -22,7 +23,7 @@ func ensureDefaultAppArmorProfile() error {
 		}
 
 		// Nothing to do.
-		if loaded {
+		if loaded && (os.Getenv("DOCKER_AA_RELOAD") == "" || os.Getenv("DOCKER_AA_RELOAD") == "0") {
 			return nil
 		}
 
