diff --git a/components/engine/pkg/listeners/group_unix.go b/components/engine/pkg/listeners/group_unix.go
index e1d8774..e27ded5 100644
--- a/components/engine/pkg/listeners/group_unix.go
+++ b/components/engine/pkg/listeners/group_unix.go
@@ -11,6 +11,7 @@ import (
 )
 
 const defaultSocketGroup = "docker"
+const unixExtrausersPath = "/var/lib/extrausers/group"
 
 func lookupGID(name string) (int, error) {
 	groupFile, err := user.GetGroupPath()
@@ -21,6 +22,12 @@ func lookupGID(name string) (int, error) {
 		return g.Name == name || strconv.Itoa(g.Gid) == name
 	})
 	if err != nil {
+		groups, err = user.ParseGroupFileFilter(unixExtrausersPath, func(g user.Group) bool {
+			return g.Name == name || strconv.Itoa(g.Gid) == name
+		})
+		if err != nil {
+			return -1, errors.Wrapf(err, "error parsing extrausers groups for %s", name)
+		}
 		return -1, errors.Wrapf(err, "error parsing groups for %s", name)
 	}
 	if len(groups) > 0 {
