diff --git a/components/engine/cmd/dockerd/daemon_unix.go b/components/engine/cmd/dockerd/daemon_unix.go
index 46ef585..647c793 100644
--- a/components/engine/cmd/dockerd/daemon_unix.go
+++ b/components/engine/cmd/dockerd/daemon_unix.go
@@ -32,6 +32,13 @@ func setDefaultUmask() error {
 }
 
 func getDaemonConfDir(_ string) string {
+	if snapData := os.Getenv("SNAP_DATA"); snapData != "" {
+		// Ubuntu Snappy 16.04+
+		return snapData + "/etc/docker"
+	} else if snapAppDataPath := os.Getenv("SNAP_APP_DATA_PATH"); snapAppDataPath != "" {
+		// Ubuntu Snappy 15.04
+		return snapAppDataPath + "/etc/docker"
+	}
 	return "/etc/docker"
 }
 
