diff --git a/vendor/src/github.com/docker/libnetwork/datastore/datastore.go b/vendor/src/github.com/docker/libnetwork/datastore/datastore.go
index c15cd62..f1ccb33 100644
--- a/vendor/src/github.com/docker/libnetwork/datastore/datastore.go
+++ b/vendor/src/github.com/docker/libnetwork/datastore/datastore.go
@@ -116,9 +116,10 @@ const (
 	LocalScope = "local"
 	// GlobalScope indicates to store the KV object in global datastore such as consul/etcd/zookeeper
 	GlobalScope   = "global"
-	defaultPrefix = "/var/lib/docker/network/files"
 )
 
+var defaultPrefix = types.FilesDir()
+
 const (
 	// NetworkKeyPrefix is the prefix for network key in the kv store
 	NetworkKeyPrefix = "network"
diff --git a/vendor/src/github.com/docker/libnetwork/sandbox_dns_unix.go b/vendor/src/github.com/docker/libnetwork/sandbox_dns_unix.go
index c8b595e..8dcf80e 100644
--- a/vendor/src/github.com/docker/libnetwork/sandbox_dns_unix.go
+++ b/vendor/src/github.com/docker/libnetwork/sandbox_dns_unix.go
@@ -16,8 +16,9 @@ import (
 	"github.com/docker/libnetwork/types"
 )
 
+var defaultPrefix = types.FilesDir()
+
 const (
-	defaultPrefix = "/var/lib/docker/network/files"
 	dirPerm       = 0755
 	filePerm      = 0644
 )
diff --git a/vendor/src/github.com/docker/libnetwork/sandbox_externalkey_unix.go b/vendor/src/github.com/docker/libnetwork/sandbox_externalkey_unix.go
index 898cedd..97a3e52 100644
--- a/vendor/src/github.com/docker/libnetwork/sandbox_externalkey_unix.go
+++ b/vendor/src/github.com/docker/libnetwork/sandbox_externalkey_unix.go
@@ -15,7 +15,8 @@ import (
 	"github.com/opencontainers/runc/libcontainer/configs"
 )
 
-const udsBase = "/var/lib/docker/network/files/"
+var udsBase = types.FilesDir() + "/"
+
 const success = "success"
 
 // processSetKeyReexec is a private function that must be called only on an reexec path
diff --git a/vendor/src/github.com/docker/libnetwork/types/types.go b/vendor/src/github.com/docker/libnetwork/types/types.go
index 44ee563..6e3a129 100644
--- a/vendor/src/github.com/docker/libnetwork/types/types.go
+++ b/vendor/src/github.com/docker/libnetwork/types/types.go
@@ -7,8 +7,26 @@ import (
 	"net"
 	"strconv"
 	"strings"
+
+	"os"
 )
 
+// XXX hack hack hack
+// see also https://github.com/docker/libnetwork/issues/747
+// see also https://github.com/docker/docker/issues/18508
+func FilesDir() string {
+	if snapData := os.Getenv("SNAP_DATA"); snapData != "" {
+		// Ubuntu Snappy 16.04+
+		return snapData + "/ln"
+	} else if snapAppDataPath := os.Getenv("SNAP_APP_DATA_PATH"); snapAppDataPath != "" {
+		// Ubuntu Snappy 15.04
+		return snapAppDataPath + "/ln"
+		// is just "/ln" because UNIX_PATH_MAX is 108 and SNAP_APP_DATA_PATH isn't exactly short :(
+		// see also https://github.com/docker/libnetwork/issues/747#issuecomment-232504914
+	}
+	return "/var/lib/docker/network/files"
+}
+
 // UUID represents a globally unique ID of various resources like network and endpoint
 type UUID string
 
